test/compilable/test21806.d

```diff
@@ -0,0 +1,24 @@
+// https://issues.dlang.org/show_bug.cgi?id=21806
+
+void main()
+{
+    ubyte[16] arr;
+    static assert(is(typeof(  fun(arr[])) == char));
+    static assert(is(typeof(funtp(arr[])) == char));
+    static assert(is(typeof(  bar(arr[])) == char));
```

Comment:
The first two make sense since `ubyte[] arr` is an exact match, but for `bar` it is a match with qualifier conversion right? So shouldn't there be partial ordering in that case?

Reply:
Ins't partial ordering applied here?

1. `ubyte[] => in ubyte[]`  is const-conversion (`MATCH.constant`)
2. `ubyte[] => in ubyte[12]` is an actual conversion (`MATCH.convert`) as stated in the linked spec

This makes (1) a better match than (2).

Reply:
Ohh, I see where I messed up. I interpreted the [match levels in the spec](https://dlang.org/spec/function.html#function-overloading) as a ranking, so I thought 2 was better than 3, but a higher match level is better.

Reply:
Well, right now in the first two cases there are partial ordering because both have same match level (exact),
in the third case `bar(in ubyte[] arr)` is `MATCH.constant` but `bar(size_t N)(in ubyte[N] arr)` is `MATCH.exact`, a little difference but the outcome of the bug is the same. However, here I've found another bug that I'm trying to fix, for example `bar(size_t N)(immutable ubyte[N] arr)` is also a match when it shouldn't.

Reply:
> However, here I've found another bug that I'm trying to fix, for example bar(size_t N)(immutable ubyte[N] arr) is also a match when it shouldn't.

Isn't that fine because `ubyte` has no indirections and the values will be copied?

Reply:
> auto b = foo(a);

I don't see any slice of `a`.

Reply:
> Isn't that fine because ubyte has no indirections and the values will be copied anyways?

Yes, I rushed that words, it was something else not relevant to this issue. I've edited my comment.